astro/src/content/blog/grammarly-proves-ciam-not-optional.mdx

```diff
@@ -0,0 +1,56 @@
+
+However, one lesson from the post is that whenever you are using OAuth tokens, you must always:
+
+* validate the signature of the token, preferably by using a library or introspection
+* validate the claims of the token
+
+This is true for any OAuth grant, including those covered by OIDC. If you want to learn more, [our article on secure signed JWTs](/articles/tokens/building-a-secure-jwt#consuming-a-jwt) goes into detail.
+
+![Verify those tokens.](/img/blogs/grammarly-oauth/verify-tokens-meme.png)
+
+## Securing Your Users Is Not Optional
+
+These breaches clearly demonstrate using proven customer identity and access management (CIAM) tools is non-optional today.
+
+Building on standards is great, but it isn't enough. A homegrown misimplementation of the standards has negative effects on your brand. Over [100 comments on HackerNews](https://news.ycombinator.com/item?id=38009291) about an article covering a security breach of yours is not great. But more importantly it impacts your users and their data. Again from the article:
+
+> "...the web editor [the compromised component] at Grammarly.com stores the data directly within Grammarly's account, which means an account takeover would give an attacker access to the victim’s stored documents. I myself am using Grammarly’s web editor for help as I've been writing this blog and writing emails, messages and documents in general."
+
+An account takeover at Grammarly, *thankfully entirely avoided per the security firm*, would have in some cases allowed an attacker to access users' private documents.
+
+Think of **your users' data being accessible** because of a mistake a developer made when implementing "Login With Facebook". I don't know about you, but this gives me the chills.
+
+CIAM is non-negotiable now. Users want to log in from multiple devices and access their profile data and all the functionality your applications offer easily and securely. They may be on a phone today and a computer tomorrow. They want to use Facebook to log in, or Google, or passkeys, or [magic links](/articles/identity-basics/magic-links). Any obstacle you put in front of them will frustrate them.
+
+But CIAM and user security isn't a core competency of most engineering teams, whether at a small company or a large one like those above. Using a standalone, hardened, secured CIAM system like FusionAuth protects your users. FusionAuth regularly undergoes outside review and penetration testing. The engineering team thinks about the security ramifications every working day (and most weekends!).
+
+The speedup in feature delivery and the outsourcing of authentication and authorization related maintenance are merely bonuses.
+
+A secure CIAM doesn't have to break the bank, either. You can use FusionAuth for free if you [download](/download) and run it yourself, or you can [buy premium features, support and hosting](/pricing) if those meet your needs. 
+
```

Comment:
you may want to say, something like: If you want to see how easy it is to use FusionAuth, checkout our quickstarts.